aplib.net-demo/Assets/Scripts/Entities/Enemy/DummyEnemy.cs

```diff
@@ -1,48 +1,42 @@
+using Entities;
 using UnityEngine;
 
 /// <summary>
 /// Dummy basic enemy class to use as target during weapons testing.
 /// It respawns after dying.
 /// </summary>
+[RequireComponent(typeof(RespawnableComponent))]
 public class DummyEnemy : AbstractEnemy
 {
     /// <summary>
-    /// The spawn area for the enemy. 
+    /// The spawn area for the enemy.
     /// </summary>
-    [SerializeField] private Area _spawnArea;
     [SerializeField] private bool _respawn = true;
-    private Bounds _spawnBounds;
+    private RespawnableComponent _respawnableComponent;
 
     /// <summary>
     /// Set the spawn bounds.
     /// This needs to happen in start, because the spawn area is not initialized yet in Awake.
     /// </summary>
     protected override void Start()
     {
-        _spawnArea ??= transform.parent.GetComponentInChildren<Area>();
-        if (_spawnArea is null)
-            Debug.LogError("No spawn area found for enemy " + name);
-        else
-            _spawnBounds = _spawnArea.Bounds;
+        _respawnableComponent = GetComponent<RespawnableComponent>();
+        _respawnableComponent.RespawnEvent += OnRespawn;
     }
 
     protected override void OnDeath(HealthComponent healthComponent)
     {
         if (_respawn)
-            Respawn();
+            _respawnableComponent.Respawn();
         else
             base.OnDeath(healthComponent);
     }
 
     /// <summary>
-    /// Respawns the enemy at a random position and resets its health.
+    /// Resets its health.
     /// </summary>
-    private void Respawn()
+    private void OnRespawn(RespawnableComponent respawnableComponent)
```

Comment:
In principe gebruik je de variabele niet dus zou ik het hier in veranderen.
```suggestion
    private void OnRespawn(RespawnableComponent _)
```